### ISSUE TITLE
Primitive exception for empty rectangle

### ISSUE THREAD

**Issue by pjcozzi** [AnalyticalGraphicsInc/cesium#2657]
Run the following in Sandcastle:
```javascript
var viewer = new Cesium.Viewer('cesiumContainer');
var scene = viewer.scene;

var instance = new Cesium.GeometryInstance({
    geometry : new Cesium.RectangleOutlineGeometry({
        rectangle : Cesium.Rectangle.fromDegrees(-100.0, 30.0, -100.0, 30.0)
    }),
    attributes : {
        color : Cesium.ColorGeometryInstanceAttribute.fromColor(Cesium.Color.WHITE)
    }
});
scene.primitives.add(new Cesium.Primitive({
    geometryInstances : instance,
    appearance : new Cesium.PerInstanceColorAppearance({
        flat : true,
        renderState : {
            lineWidth : Math.min(2.0, scene.maximumAliasedLineWidth)
        }
    })
}));
```
Note that the rectangle's `west === east` and `south === north`.
![image](https://cloud.githubusercontent.com/assets/782098/7274429/a73cc7ce-e8ca-11e4-81dc-9fb1833be4d8.png)
This is not an exceptional case.  We should just render nothing.  I suspect there is a similar issue in `RectangleGeometry` and probably most geometries.

@bagnell can you have a look after we finish billboards/labels on terrain?

**Comment by TomPed** [comment]
If you change this:
```javascript
rectangle : Cesium.Rectangle.fromDegrees(-100.0, 30.0, -100.0, 30.0)
```
to this:
```javascript
rectangle : Cesium.Rectangle.fromDegrees()
```
I believe that you get the same error. So should we also not allow the parameters to be optional? Or should I just add and if check: `if ((north === south) || (east === west))` and handle this case separately?

**Comment by pjcozzi** [comment]
Leave `Rectangle.fromDegrees` as is, and do the check in `RectangleOutlineGeometry` and `OutlineGeometry` (and most likely through the pipeline into `Primitive`, which should now not assume the input geometry has any vertices.

We should check the other geometries too, e.g., if all sides of a box have length zero, etc.

**Comment by TomPed** [comment]
Gotcha, thanks.

**Comment by hpinkos** [comment]
Fixed in #3599
